### ISSUE TITLE
Feature/adjustment request: "plague" promotion functionality

### ISSUE THREAD

**Issue by Tekamthi** [LoneGazebo/Community-Patch-DLL#10747]
The promotions table contains a field, 'PlagueChance', that when combined with another that assigns a specific promotion as plague, allows a unit to give a promotion to another via melee combat.  

Request 1: would be preferable as modder to have this field divided into 'PlagueAttackChance' and 'PlagueDefendChance'.  ie allow plague promotions to be conferred only by an attacker or only by defender, or both.

Request 2: the plagues ignore the promotion eligibility definitions in UnitPromotions_UnitCombats -- would give modder greater control if these were respected ie plague only applied to units eligible in this table.

**Comment by azum4roll** [comment]
For reference, plagues are currently inflicted on the attacker if defending against a melee attack as a melee unit, and not otherwise.

**Comment by Tekamthi** [comment]
Yes plagues are melee only in all testing I've done with this.  If any devs end up looking into it, I suppose we might want to consider whether they should apply via ranged attack as well but I am indifferent on this point personally (though it should work fine gameplay wise); melee only but divided between attack and defense is still desirable functionality

**Comment by azum4roll** [comment]
Plagues are also inflicted by ranged units on attacking only (see Waraq'Ak).

**Comment by Tekamthi** [comment]
my mistake, been a while since I tried adding to ranged unit

**Comment by axatin** [comment]
The first request is now implemented. For the second one you can give the unit types you want to exclude a promotion that blocks the plague.
